apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -644,10 +645,9 @@ private void Open(Action transactionEnlister)
         /// <returns>A task which will complete when the session pool has reached its minimum size.</returns>
         public async Task WhenSessionPoolReady(CancellationToken cancellationToken = default)
         {
-            DatabaseName databaseName = Builder.DatabaseName;
-            GaxPreconditions.CheckState(databaseName != null, $"{nameof(WhenSessionPoolReady)} cannot be used without a database.");
             await OpenAsync(cancellationToken).ConfigureAwait(false);
-            await _sessionPool.WhenPoolReady(databaseName, cancellationToken).ConfigureAwait(false);
+            var sessionPoolSegmentKey = GetSessionPoolSegmentKey(nameof(WhenSessionPoolReady));
```

Comment:
Move this to before the call to OpenAsync.
We don't want to attempt to opne the connection if there's no database specified. Connection opening will fail, but the error will be off.

Do this here and in all the others.